### PR TITLE
Add imports key for remote client

### PR DESCRIPTION
**Pull request by Emantor**
TODO:
- [x] add Tests

### CODE REVIEW

File: labgrid/environment.py [hunk 1]
```diff
@@ -23,6 +24,9 @@ def __attrs_post_init__(self):
             raise NoConfigFoundError(
                 "{} is not a valid yaml file".format(self.config_file)
             )
+        for user_file in self.config.get_imports():
+            importlib.import_module(user_file)
+            importlib.invalidate_caches()
```
Comment (jluebbe) [inline labgrid/environment.py]: We don't need invalidate_caches, as the file should already exist when starting labgrid.

File: labgrid/environment.py [hunk 2]
```diff
@@ -23,6 +24,9 @@ def __attrs_post_init__(self):
             raise NoConfigFoundError(
                 "{} is not a valid yaml file".format(self.config_file)
             )
+        for user_file in self.config.get_imports():
+            importlib.import_module(user_file)
```
Comment (jluebbe) [inline labgrid/environment.py]: Don't we need to resolve the path relative to the environment file, before passing it to importlib? We should use the same behavior as images or normal paths in the config.

File: labgrid/config.py [hunk 3]
```diff
@@ -128,3 +128,11 @@ def set_option(self, name, value):
...
     def get_targets(self):
         return self.data.get('targets', {})
+
+    def get_imports(self):
+        """Helper function that returns the list of all imports
+
+        Returns:
+            list: List of files which should be imported
+        """
+        return self.data.get('imports', [])
```
Comment (jluebbe) [inline labgrid/config.py]: Resolve the path here to be consistent with the other config APIs.

File: labgrid/environment.py [hunk 4]
```diff
@@ -23,6 +24,15 @@ def __attrs_post_init__(self):
             raise NoConfigFoundError(
                 "{} is not a valid yaml file".format(self.config_file)
             )
+        for user_file in self.config.get_imports():
+            from importlib.machinery import SourceFileLoader
+            import sys
+
+            file_path = self.config.resolve_path(user_file)
```
Comment (jluebbe) [inline labgrid/environment.py]: This is no longer needed when the Config resolves the paths.

File: man/labgrid-device-config.1 [hunk 5]
```diff
@@ -87,6 +87,10 @@ Path to the mxs\-usb\-loader binary
 .B \fBimx\-usb\-loader\fP
 Path to the imx\-usb\-loader binary
 .UNINDENT
+.SH IMPORTS
+.sp
+The \fBimports\fP key is a list of files which are imported by the environment
+after loading the configuration.
```
Comment (jluebbe) [inline man/labgrid-device-config.1]: Add a note that paths relative to the config file are supported.

File: tests/test_environment.py [hunk 6]
```diff
@@ -50,3 +50,26 @@ def test_instance_invalid_yaml(self, tmpdir):
...
+        targets:
+          main:
+            drivers: {}
+        imports:
+          - {}
+        """.format("{}",str(i))
+        )
+        e = Environment(str(p))
+        assert (isinstance(e, Environment))
+        assert "myimport" in sys.modules
```
Comment (jluebbe) [inline tests/test_environment.py]: Also check that the Test class exists.

**Comment by Emantor** [comment]
Adressed your comments, should be ready to merge.

**Pull request merged**